internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/preprocessor/PreProcessor.java

```diff
@@ -0,0 +1,171 @@
+     * Product pack can be downloaded from any resource. Currently it supports GitHub.
+     * Package.json files can be downloaded from github repo as raw materials.
+     *
+     * @param productDtoList Product list
+     * @throws DownloaderException exception occurred while downloading files.
+     */
+    private void downloadResources(List<Product> productDtoList) throws DownloaderException,
+            ApiInvokerException, ConfigParserException {
+        ConfigParser.parseGitAccessToken();
+        productFileMapper = new HashMap<>();
+        String filename;
+        for (Product productDto : productDtoList) {
+            //create directory for current product
+            File currentProductDir = new File(productRootDirectory.getAbsolutePath() + File.separator +
+                    productDto.getProductRepoName());
+            createDirectory(currentProductDir);
+
+            List<String> zipFileList = new ArrayList<>();
+            //download from git repository
+            if (productDto.getInputSourceType().equals(JSScannerConstants.GIT)) {
+                ResourceDownloader gitDownloader = new GitDownloader();
+                startDownload(productDto, currentProductDir, zipFileList, gitDownloader);
+            }
+            if (productDto.getInputSourceType().equals(JSScannerConstants.ATUWA)) {
+                ResourceDownloader atuwaDownloader = new AtuwaDownloader();
+                ConfigParser.parseAtuwaUrl();
+                startDownload(productDto, currentProductDir, zipFileList, atuwaDownloader);
+            }
+            //unzip downloaded product pack
+
+            if (zipFileList.size() > 0) {
+                for (String filePath : zipFileList) {
+                    filename = new File(filePath).getName();
+                    productFileMapper.put(filename.substring(0, (filename.length() -
+                                    JSScannerConstants.FILE_SUFFIX_LENGTH)),
+                            currentProductDir.getAbsolutePath());
+                    log.info("[JS_SEC_DAILY_SCAN]  " + " Unzipped file : " + zipFileList);
+                }
+            }
+        }
+    }
+
+    /**
+     * This method is common entry point to start downloading product packs.
+     *
+     * @param productDto         object which contains details of product.
+     * @param currentProductDir  Current product root directory.
+     * @param zipFileList        List to store downloaded product packs zip file list.
+     * @param resourceDownloader Resource downloader. This object indicates the type of source where we download
+     *                           product packs. Eg : Github, Atuwa.
+     * @throws ApiInvokerException Exception occurred when calling API.
+     * @throws DownloaderException Exception occurred when downloading files.
+     */
+    private void startDownload(Product productDto, File currentProductDir, List<String> zipFileList, ResourceDownloader
```

Comment:
The name 'startDownload' gives the idea that the download will be only started but not completed. Is that the case?